src/widgets/graph-widget/hooks/use-data.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+import { StateSelector } from 'zustand';
+import { EventBusState, useEventBus } from '@wuespace/telestion-client-core';
+
+import { ChartConnection, DataSample } from '../model';
+import { useCallbacks } from './use-callbacks';
+
+// event bus selector
+const selector: StateSelector<
+	EventBusState,
+	EventBusState['eventBus']
+> = state => state.eventBus;
+
+export function useData(connections: ChartConnection[]) {
+	const eventBus = useEventBus(selector);
+
+	const [data, setData] = useState<DataSample[]>([]);
+	const callbacks = useCallbacks(connections, setData);
+
+	if (!eventBus) {
+		throw new TypeError(
+			'There is no eventbus instance to subscribe to. ' +
+				'Please check if the eventbus was created and try again.'
+		);
+	}
+
+	useEffect(() => {
+		// register all callbacks
+		callbacks.forEach(([channel, callback]) =>
+			eventBus.registerHandler(channel, callback)
+		);
+		// unregister all callbacks
+		return () =>
+			callbacks.forEach(([channel, callback]) =>
+				eventBus.unregisterHandler(channel, callback)
+			);
+	}, [callbacks, connections, eventBus]);
```

Comment:
Is it possible (right now, of the top of my head, I don't know if it is, so this is simply a question) to just check if there's an `eventBus` instance in the effect?

```suggestion
	useEffect(() => {
		if (eventBus) {
			// register all callbacks
			callbacks.forEach(([channel, callback]) =>
				eventBus.registerHandler(channel, callback)
			);
			// unregister all callbacks
			return () =>
				callbacks.forEach(([channel, callback]) =>
					eventBus.unregisterHandler(channel, callback)
				);
		}
	}, [callbacks, connections, eventBus]);
```

Reply:
Good point!
During the implementation, I thought, I stick to the abstraction hook behaviours ([TelestionTeam/telestion-client#use-channel.ts](https://github.com/TelestionTeam/telestion-client/blob/main/packages/telestion-client-core/src/hooks/abstractions/event-bus-abstractions/use-channel.ts)) from the core package. But your question is still valid.

Reply:
Is commited with https://github.com/TelestionTeam/telestion-rocketsound-psc/pull/49/commits/9edfabe4dccd7a21d58d67dbd8f24ae6f910542d.
Thanks! :+1: